owncloudApp/src/main/java/com/owncloud/android/presentation/files/createshortcut/CreateShortcutDialogFragment.kt

```diff
@@ -0,0 +1,154 @@
+        super.onViewCreated(view, savedInstanceState)
+        binding.apply {
+            handleInputsUrlAndFileName()
+            yesButton.setOnClickListener {
+                if (isCreateShortcutButtonEnable) {
+                    createShortcutListener.createShortcutFileFromApp(
+                        fileName = createShortcutDialogNameFileValue.text.toString(),
+                        url = formatUrl(createShortcutDialogUrlValue.text.toString()),
+                    )
+                    dialog?.dismiss()
+                }
+            }
+            cancelButton.setOnClickListener {
+                dialog?.dismiss()
+            }
+        }
+    }
+
+    private fun formatUrl(url: String): String {
+        var formattedUrl = url
+        if (!url.startsWith("http://") && !url.startsWith("https://")) {
+            formattedUrl = "http://$url"
+        }
+        return formattedUrl
+    }
+
+    private fun handleInputsUrlAndFileName() {
+        binding.createShortcutDialogNameFileValue.doOnTextChanged { fileNameValue, _, _, _ ->
+            fileNameValue?.let {
+                hasForbiddenCharacters = FORBIDDEN_CHARACTERS.any { fileNameValue.contains(it) }
+                hasMaxCharacters = fileNameValue.length >= MAX_FILE_NAME
+                isValidFileName = fileNameValue.isNotBlank() && !hasForbiddenCharacters && !hasMaxCharacters
+                handleNameRequirements(hasForbiddenCharacters, hasMaxCharacters)
+                updateCreateShortcutButtonState()
+            }
+        }
+        binding.createShortcutDialogUrlValue.doOnTextChanged { urlValue, _, _, _ ->
+            urlValue?.let {
+                hasEmptyValue = urlValue.contains(" ")
+                isValidUrl = urlValue.isNotBlank() && !hasEmptyValue
+                handleUrlRequirements(hasEmptyValue)
+                updateCreateShortcutButtonState()
+            }
+        }
+    }
+
+    private fun updateCreateShortcutButtonState() {
+        isCreateShortcutButtonEnable = isValidFileName && isValidUrl
+        enableYesButton(isCreateShortcutButtonEnable)
+    }
+
+    private fun handleNameRequirements(hasForbiddenCharacters: Boolean, hasMaxCharacters: Boolean) {
+        binding.createShortcutDialogNameFileLayout.apply {
+            error = when {
+                hasMaxCharacters -> getString(R.string.uploader_upload_text_dialog_filename_error_length_max, MAX_FILE_NAME)
+                hasForbiddenCharacters -> getString(R.string.filename_forbidden_characters)
+                else -> null
+            }
+        }
+    }
+
+    private fun handleUrlRequirements(hasSpace: Boolean) {
+        binding.createShortcutDialogUrlLayout.apply {
+            if (hasSpace) {
+                error = getString(R.string.create_shortcut_dialog_url_error_no_spaces)
+            } else {
+                error = null
+            }
+        }
+    }
+
+    private fun enableYesButton(enable: Boolean) {
```

Comment:
This method should set the listener as well when `enable` is true, or set it to `null` when `enable` is false

Reply:
Done